code/datums/status_effects/xeno_buffs.dm

```diff
@@ -778,7 +780,8 @@
 	if(!isxeno(owner))
 		return FALSE
 	var/mob/living/carbon/xenomorph/X = owner
-	X.soft_armor = X.soft_armor.modifyAllRatings(SENTINEL_DRAIN_SURGE_ARMOR_MOD)
+	for(var/stat in X.resist_modifiers)
```

Comment:
This seems like an uncouth way to handle this.

Reply:
Why's that?

Reply:
Usually for handling armor datums we use associated helper procs, but this's just an assoc list you're iterating through instead. IDK, just feels off stylistically.

This does seem like just a second set of armor for yet another sunder workaround, but it's not like I have a better solution.